cdap-watchdog-api/src/main/java/io/cdap/cdap/api/metrics/MetricValue.java

```diff
@@ -120,7 +122,17 @@ public String toString() {
    */
   public long[] getAllBucketCounts() {
     // TODO Move to helper class if serialization uses get methods
-    throw new UnsupportedOperationException();
+    long[] allBucketCounts = new long[NUM_FINITE_BUCKETS + 2];
+    long mask = 1;
+    int index = 0;
+    for (int i = 0; i < allBucketCounts.length; i++) {
+      if ((bucketMask & mask) >= 1) {
+        allBucketCounts[i] = bucketCounts[index];
```

Comment:
IntelliJ gave a potential NPE warning for the usage of `bucketCounts`. You probably should check the for it to guard against future bug.

Reply:
Thanks. Added type checks which I missed earlier. Also add a defensive null check for bucketCounts.